arcgis-ios-sdk-samples/Layers/List KML contents/ListKMLContentsSceneViewController.swift

```diff
@@ -0,0 +1,202 @@
+// Copyright 2018 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class ListKMLContentsSceneViewController: UIViewController {
+    
+    @IBOutlet weak var sceneView: AGSSceneView? {
+        didSet {
+            sceneView?.scene = scene
+        }
+    }
+    
+    private let scene: AGSScene = {
+        
+        // initialize scene with labeled imagery
+        let scene = AGSScene(basemapType: .imageryWithLabels)
+        
+        // create an elevation source and add it to the scene's surface
+        let elevationSourceURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+        let elevationSource = AGSArcGISTiledElevationSource(url: elevationSourceURL)
+        scene.baseSurface?.elevationSources.append(elevationSource)
+        
+        return scene
+    }()
+    
+    var kmlDataset: AGSKMLDataset? {
+        didSet {
+            guard kmlDataset != oldValue,
+                let kmlDataset = kmlDataset else {
+                    return
+            }
+            
+            // clear any existing layers
+            scene.operationalLayers.removeAllObjects()
+            
+            // create a layer to display the dataset
+            let kmlLayer = AGSKMLLayer(kmlDataset: kmlDataset)
+            // add the kml layer to the scene
+            scene.operationalLayers.add(kmlLayer)
+        }
+    }
+    var node: AGSKMLNode? {
+        didSet {
+            guard kmlDataset != oldValue,
+                let node = node else{
+                return
+            }
+            // set the title so the name will appear in the navigation bar
+            title = node.name
+                
+            // set the viewpoint based on the new node
+            setSceneViewpoint(for: node)
+        }
+    }
+    
+    //MARK: - Viewpoint
+    
+    /// Sets the viewpoint of the scene based on the node, or hides the node
+    /// if a viewpoint cannot be determined.
+    private func setSceneViewpoint(for node:AGSKMLNode){
+        
+        // load the view before setting a viewpoint
+        loadViewIfNeeded()
```

Comment:
I should have noticed this sooner, but unless I'm missing something, the view doesn't need to be loaded here. Should this be moved to after dispatching back to the main queue, right before you unwrap `self.sceneView`?

Speaking of unwrapping the scene view, I'm curious: why did you make it an optional rather than an IUO? As you know, IUO is kind of the standard for `IBOutlet`s. In addition, it looks like you're rightly making sure that the view is loaded before accessing `self.sceneView`, so using an IUO would prevent you from having to explicitly unwrap it yourself.